src/konserve_jdbc/core.clj

```diff
@@ -106,157 +107,150 @@
-                                     :or {table default-table}
-                                     :as params}]
+                (go-try- (let [res' (jdbc/execute! connection
+                                                   [(str "SELECT id FROM " table ";")]
+                                                   {:builder-fn rs/as-unqualified-lower-maps})]
+                           (map :id res'))))))
+
+(defn connect-store [db-spec & {:keys [table opts]
+                                :or {table default-table}
+                                :as params}]
   (let [complete-opts (merge {:sync? true} opts)
-        datasource (jdbc/get-datasource db-spec)
-        backing (JDBCTable. db-spec datasource table)
-        config (merge {:table              table
-                       :opts               complete-opts
-                       :config             {:sync-blob? false
+        connection (jdbc/get-connection db-spec)
+        db-spec (if (:dbtype db-spec)
+                  db-spec
+                  (assoc db-spec :dbtype (:subprotocol db-spec)))
+        backing (JDBCTable. db-spec connection table)
+        config (merge {:opts               complete-opts
+                       :config             {:sync-blob? true
                                             :in-place? true
                                             :lock-blob? true}
                        :default-serializer :FressianSerializer
                        :compressor         null-compressor
                        :encryptor          null-encryptor
                        :buffer-size        (* 1024 1024)}
-                      (dissoc params :table :opts :config))
-        db-type (or (:dbtype db-spec) (:subprotocol db-spec))]
-    (when-not db-type
+                      (dissoc params :opts :config))]
+    (when-not (:dbtype db-spec)
       (throw (ex-info ":dbtype must be explicitly declared" {:options dbtypes})))
-    (when-not (supported-dbtypes db-type)
-      (warn "Unsupported database type " db-type
+    (when-not (supported-dbtypes (:dbtype db-spec))
+      (warn "Unsupported database type " (:dbtype db-spec)
             " - full functionality of store is only guaranteed for following database types: "  supported-dbtypes))
     (connect-default-store backing config)))
 
+(defn release
+  "Must be called after work on database has finished in order to close connection"
+  [store env]
+  (async+sync (:sync? env) *default-sync-translation*
+              (go-try- (.close ^Connection (:connection ^JDBCTable (:backing store))))))
+
 (defn delete-store [db-spec & {:keys [table opts] :or {table default-table}}]
   (let [complete-opts (merge {:sync? true} opts)
-        datasource (jdbc/get-datasource db-spec)
-        backing (JDBCTable. db-spec datasource table)]
+        connection (jdbc/get-connection db-spec)
+        backing (JDBCTable. db-spec connection table)]
     (-delete-store backing complete-opts)))
 
 (comment
```

Comment:
At some point I would remove the comments out of the core namespaces. But for now it's fine.